### PR TITLE
make recovery more general in stress module and labfront loader

### PR DESCRIPTION
**Pull request by Vaeliss**
Recovery calculation in the stress module should be more general, the loading of the metric of recovery is going to be specific for the different types of loaders. 

In this commit the garmin loader of labfront is taking care of appropriate loading of recovery defined as:
(body battery waking - body battery bedtime)/(100 - body battery bedtime)

This is a temporary update to the calculation of the recovery statistics (with a considerable 10x speed up in computation time). The stress module will have to undergo several changes in successive updates.

### CODE REVIEW

File: src/pywearable/loader/labfront/loader.py [hunk 1]
```diff
@@ -2432,3 +2432,110 @@ def load_stress(
...
+                        how="inner",
+                        on=constants._CALENDAR_DATE_COL,
+                        suffixes=["Bedtime", "Wakeup"],
+                    )
+                    sleep_timestamps[constants._RECOVERY_RECOVERY_COL] = (
+                        sleep_timestamps["bodyBatteryWakeup"]
+                        - sleep_timestamps["bodyBatteryBedtime"]
+                    ) / (100 - sleep_timestamps["bodyBatteryBedtime"])
+                    # make sure recovery isn't negative, set it to zero if it is
+                    sleep_timestamps[constants._RECOVERY_RECOVERY_COL] = np.maximum(sleep_timestamps[constants._RECOVERY_RECOVERY_COL],0)
```
Comment (dado93) [inline src/pywearable/loader/labfront/loader.py:2530]: Is the np.maximum operation more efficient than doing:
`sleep_timestamps.loc[sleep_timestamps[constants._RECOVERY_RECOVERY_COL]<0,constants._RECOVERY_RECOVERY_COL]=0`?

File: src/pywearable/loader/labfront/loader.py [hunk 2]
```diff
@@ -2432,3 +2432,110 @@ def load_stress(
...
+                    sleep_timestamps = sleep_timestamps.reset_index(drop=True)
+                    return sleep_timestamps.loc[
+                        :,
+                        [
+                            constants._CALENDAR_DATE_COL,
+                            constants._RECOVERY_RECOVERY_COL,
+                            "bodyBatteryBedtime",
```
Comment (dado93) [inline src/pywearable/loader/labfront/loader.py:2537]: Probably better to have them as constants in the constants module

File: src/pywearable/loader/labfront/loader.py (hunk 1 above)
Comment (Vaeliss) [reply src/pywearable/loader/labfront/loader.py:2530]: They're both efficient but np.maximum in a timeit test (with the test repeated 10000 times) seems to be about twice as fast. (average execution time 0.0003313874006271362s for np.maximum vs 0.0006034422397613525s for the other approach, for a pd.DataFrame with 24000 rows).

File: src/pywearable/loader/labfront/loader.py (hunk 2 above)
Comment (Vaeliss) [reply src/pywearable/loader/labfront/loader.py]: Agreed. Pull request was updated with [commit 41ea60a](https://github.com/dado93/pywearable/commit/41ea60a5f82f882beee7161cb1ab03b8792b11e3)

**Pull request merged**